### PR TITLE
Making Requestlog write async

### PR DESCRIPTION
**Pull request by pavanab4u**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: request-log/src/main/scala/com/yahoo/maha/log/KafkaMahaRequestLogWriter.scala [hunk 1]
```diff
@@ -90,7 +90,10 @@ class KafkaMahaRequestLogWriter(kafkaRequestLoggingConfig: KafkaRequestLoggingCo
   }
 
   def write(proto: MahaRequestProto) : Unit = {
-    writeMahaRequestProto(proto)
+    implicit val ec: scala.concurrent.ExecutionContext = scala.concurrent.ExecutionContext.global
```
Comment (patelh) [inline request-log/src/main/scala/com/yahoo/maha/log/KafkaMahaRequestLogWriter.scala:93]: This feels hacky, why not create another method which takes in a executoncontext, e.g.
```
def writeAsync(proto: MahaRequestProto, ec: ExecutionContext): Unit
```
This way you can control the execution context and what to do when the pool is full.

Comment (patelh) [reply request-log/src/main/scala/com/yahoo/maha/log/KafkaMahaRequestLogWriter.scala:93]: Also, I don't see why it isn't async already.  The writeMahaRequestProto method already returns a future.  It's already async.

Comment (pranavbhole) [reply request-log/src/main/scala/com/yahoo/maha/log/KafkaMahaRequestLogWriter.scala:93]: it is actually problem of the kafka producer library, when kafka server is not available, it blocks the call wait for the response from server for "max block ms" this blocks the response in api. Thus wrapping it up in the another thread.

Comment (patelh) [reply request-log/src/main/scala/com/yahoo/maha/log/KafkaMahaRequestLogWriter.scala:93]: Yea, but the blocking behavior only occurs if the local queue is full.  All you are doing by using another thread pool is passing the blocking behavior into another thread pool.  The call in the other thread pool will block and soon that pool be exhausted and you'll have a bunch of submitted tasks in its queue.  Effectively causing a memory leak.  The max size of forkjoin pool is 64 million.

Comment (patelh) [reply request-log/src/main/scala/com/yahoo/maha/log/KafkaMahaRequestLogWriter.scala:93]: Ideally configure kafka client to timeout earlier or size the queue such that once queue is full it starts dropping messages.

Comment (pranavbhole) [reply request-log/src/main/scala/com/yahoo/maha/log/KafkaMahaRequestLogWriter.scala:93]: fixing

**Pull request merged**